File: java/csp/src/main/java/sudoku/example_scraping.java
package sudoku;

import java.util.AbstractMap;
import java.util.Map;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class example_scraping {
    public static void main(String[] args) {
        Map<String, String> sudokus = Stream.
                of(new AbstractMap.SimpleEntry<>("easy", ""),
                        new AbstractMap.SimpleEntry<>("medium", ""),
                        new AbstractMap.SimpleEntry<>("hard", ""),
                        new AbstractMap.SimpleEntry<>("evil", "")).
                collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
        try (Fetcher fetcher = new Fetcher()) {
            // Get sudokus
            sudokus.keySet().forEach(sudoku -> {
                sudokus.put(sudoku, fetcher.fetch_sudoku(sudoku.toUpperCase()));
            });

            // Solve sudokus
            sudokus.keySet().forEach(sudoku -> {
                Sudoku s = new Sudoku(sudokus.get(sudoku), "mrv", "lcv");
                long before = System.currentTimeMillis();
                if(s.backtracking_search()) {
                    System.out.println(s.pretty_print());
                } else {
                    System.out.println("No solution :(");
                }
                double now = (System.currentTimeMillis() - before) / 1000.0;
                System.out.println(String.format("RAW SUDOKU [%s]: %s (%.3fs)\n", sudoku.toUpperCase(),
                        sudokus.get(sudoku), now));
            });
        }
    }
}

File: python/csp_solver/src/sudoku/example_scraping.py
from csp_solver.src.sudoku.Fetcher import Fetcher
from csp_solver.src.sudoku.Sudoku import Sudoku

if __name__ == '__main__':
    # We get one sudoku per class
    sudokus = {
        'easy': '',
        'medium': '',
        'hard': '',
        'evil': ''
    }

    f = Fetcher()
    for sudoku in sudokus:
        sudokus[sudoku] = f.fetch_sudoku(sudoku.upper())

    # Solve sudokus
    for sudoku in sudokus:
        s = Sudoku(sudokus[sudoku], 'mrv', 'lcv')
        print('RAW SUDOKU [%s]: %s\n' % (sudoku.upper(), sudokus[sudoku]))
        if s.backtracking_search():
            print(s.pretty_print())
        else:
            print('No solution :(')
